stl/inc/format

```diff
@@ -3124,14 +3173,125 @@ _NODISCARD const _CharT* _Measure_string_prefix(const basic_string_view<_CharT>
     return _Pfx_iter._Position();
 }
 
+_NODISCARD constexpr bool _Is_printable(char32_t _Code_point) {
+    return _Has_property(_Code_point, __printable_ranges, _STD end(__printable_ranges));
+}
+
+_NODISCARD constexpr bool _Is_grapheme_extend(char32_t _Code_point) {
+    return _Has_property(_Code_point, _Grapheme_Extend_ranges, _STD end(_Grapheme_Extend_ranges));
+}
+
+template <class _CharT, class _OutputIt>
+_NODISCARD _OutputIt _Write_escaped(_OutputIt _Out, basic_string_view<_CharT> _Value, char _Delim) {
+    auto _First        = _Value.data();
+    const auto _Last   = _First + _Value.size();
+    const auto& _Codec = _Get_fmt_codec<_CharT>();
+
+    _STL_INTERNAL_CHECK(_Delim == '"' || _Delim == '\'');
+    *_Out++ = static_cast<_CharT>(_Delim);
+
+    bool _Escape_grapheme_extend = true;
+
+    const auto _Write_backslash_and = [&](char _Ch) {
+        *_Out++                 = static_cast<_CharT>('\\');
+        *_Out++                 = static_cast<_CharT>(_Ch);
+        _Escape_grapheme_extend = true;
+    };
+
+    const auto _Write_hex = [&](uint32_t _Val) {

```

Comment:
ditto we could probably get away with just calling _Write_fmt with the right specs here.

Reply:
I decided to simply inline this lambda. We don't need the flexibility of `_Fmt_write`.